src/components/swap/ConfirmSwapModal.tsx

```diff
@@ -124,6 +127,13 @@ export default function ConfirmSwapModal({
     }
   }, [allowance, previousPermitNeeded])
 
+  const swapFailed = useMemo(() => Boolean(swapError) && !didUserReject(swapError), [swapError])
+  useEffect(() => {
+    if (swapError && !swapFailed && ConfirmModalState.PENDING_CONFIRMATION === confirmModalState) {
+      setConfirmModalState(ConfirmModalState.REVIEWING)
+    }
+  }, [confirmModalState, swapError, swapFailed])
```

Comment:
Can this be put somewhere where it is triggered imperatively instead of as part of a `useEffect`?

Reply:
yes, updated